// decky-frontend-lib/dist/deck-components/static-classes.js
import { findModule } from '../webpack';
export const quickAccessMenuClasses = findModule((mod) => typeof mod === 'object' && mod?.Title && mod?.QuickAccessMenu && mod?.BatteryDetailsLabels);
/**
 * @depreciated please use quickAccessMenuClasses instead
 */
export const staticClasses = quickAccessMenuClasses;
export const scrollPanelClasses = findModule((mod) => typeof mod === 'object' && mod?.ScrollPanel);
/**
 * @depreciated please use scrollPanelClasses instead
 */
export const scrollClasses = scrollPanelClasses;
// TODO refactor to use class mapper
export const gamepadDialogClasses = findModule((mod) => typeof mod === 'object' && mod?.GamepadDialogContent);
export const quickAccessControlsClasses = findModule((mod) => typeof mod === 'object' && typeof mod?.PanelSection === 'string' && mod?.PanelSection);
export const updaterFieldClasses = findModule((mod) => typeof mod === 'object' && mod?.OOBEUpdateStatusContainer);
export const playSectionClasses = findModule((mod) => typeof mod === 'object' && mod?.Container);
export const gamepadSliderClasses = findModule((mod) => typeof mod === 'object' && mod?.SliderControlPanelGroup);
export const appDetailsHeaderClasses = findModule((mod) => typeof mod === 'object' && mod?.TopCapsule);
export const appDetailsClasses = findModule((mod) => typeof mod === 'object' && mod?.HeaderLoaded);
export const gamepadUIClasses = findModule((mod) => typeof mod === 'object' && mod?.BasicUiRoot);
export const gamepadTabbedPageClasses = findModule((mod) => typeof mod === 'object' && mod?.GamepadTabbedPage);
export const gamepadContextMenuClasses = findModule((mod) => typeof mod === 'object' && mod?.BasicContextMenuModal);
export const achievementClasses = findModule((mod) => typeof mod === 'object' && mod?.AchievementListItemBase);
export const mainMenuAppRunningClasses = findModule((mod) => typeof mod === 'object' && mod?.MainMenuAppRunning);
export const basicAppDetailsSectionStylerClasses = findModule((mod) => typeof mod === 'object' && mod?.AppDetailsRoot);
